src/distribution/lognormal.rs

```diff
@@ -0,0 +1,223 @@
+use distribution;
+use source::Source;
+
+/// A lognormal distribution.
+#[derive(Clone, Copy)]
+pub struct Lognormal {
+    mu: f64,
+    sigma: f64,
+}
+
+impl Lognormal {
+    /// Create a lognormal distribution with location `mu` and scale `sigma`.
+    ///
+    /// It should hold that `sigma > 0`.
+    #[inline]
+    pub fn new(mu: f64, sigma: f64) -> Self {
+        should!(sigma > 0.0);
+        Lognormal { mu: mu, sigma: sigma }
+    }
+
+    /// Return the location parameter.
+    #[inline(always)]
+    pub fn mu(&self) -> f64 { self.mu }
+
+    /// Return the scale parameter.
+    #[inline(always)]
+    pub fn sigma(&self) -> f64 { self.sigma }
+}
+
+impl Default for Lognormal {
+    #[inline]
+    fn default() -> Self {
+        Lognormal::new(0.0, 1.0)
+    }
+}
+
+impl distribution::Continuous for Lognormal {
+    #[inline]
+    fn density(&self, x: f64) -> f64 {
+        use std::f64::consts::PI;
+        should!(x > 0.0);
+        let &Lognormal { mu, sigma } = self;
+        (-(x.ln() - mu).powi(2) / (2.0 * sigma * sigma)).exp() / (x * sigma * (2.0 * PI).sqrt())
+    }
+}
+
+impl distribution::Distribution for Lognormal {
+    type Value = f64;
+
+    fn distribution(&self, x: f64) -> f64 {
+        distribution::Gaussian::new(self.mu, self.sigma).distribution(x.ln())
```

Comment:
I think we should be consistent. Given zero, `density` will panic, but `distribution` will not. So we should either panic in both cases or accept zero as a valid input in both cases. Maybe we should actually consider zero as a valid input. Technically speaking, Gaussian distributions are not defined for +/- infinity, but we accept them anyway. Perhaps we should treat zero here in the same way.